### PR TITLE
Add CONTRIBUTING.md

### LINKED ISSUE #3014: Create a separate commit for builds, much easier to merge/rebase
@shockey @bodnia @webron 
The `/dist` folder often comes into merge conflicts, making feature branches difficult to keep clean and up to date.

Is it possible to enforce a rule ( "enforce "), whereby the `build` happens on its own commit? That way an interactive rebase becomes a smooth operation ( since we just ignore all those build commits, which are useless in a rebase ). The ends are still met, where we have dist built, and in the PR it'll merge happily

Ultimately just our internal practice and a mention in the README?

Food for thought, I can add the note in the readme - if this makes sense.

### PR DESCRIPTION
**Pull request by shockey**
Fixes #3014.

Here's my proposal for a Contributing Guidelines document.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+## Contributing to Swagger-UI
+
+We love contributions from our community of users! This document explains our guidelines and workflows. Please take care to follow them, as it helps us keep things moving smoothly.
+
+#### Environment setup
+
+0. Install Node.js (4 or newer) and npm (3 or newer).
+1. Fork Swagger-UI, and clone your fork.
```
Comment (ponelat) [inline CONTRIBUTING.md]: double clone?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+- Break your commits into logical atomic units. Well-segmented commits make it _much_ easier for others to step through your changes.
+- Limit your subject (first) line to 50 characters (GitHub truncates more than 70).
+- Provide a body if you'd like to explain your commit in detail.
+- Separate the subject from the body with a blank line, for readability.
+- Capitalize the beginning of your subject line.
+- Do not end the subject line with a period.
+- Use the imperative mood in your subject lines, as if you were giving the code an order in your subject line.
+  - This mimics what Git does for you automatically: "Merge develop", "Revert f570ffc", etc
+  - Simple trick... your subject line should complete this sentence: `If applied, this commit will [your subject line].`
```
Comment (ponelat) [inline CONTRIBUTING.md]: This is an awesome tip.
Would it make sense to use your tip, plus maybe two other points? I think the guide may be larger than needed, IMO.

File: CONTRIBUTING.md (hunk 1 above)
Comment (shockey) [inline CONTRIBUTING.md]: @ponelat intended to write: Make a fork of Swagger-UI on GitHub, then clone your fork to your machine.

Will change for clarity, thx!

**Comment by EWhite613** [comment]
Shouldn't this file be located in `.github/CONTRIBUTING.md` and you'll get the warning above a PR when ur about to create that reminds you to `Before you create a pull request, please review [the guidelines for contributing](link) to this repository.`

**Comment by shockey** [comment]
@EWhite613, looks like that's how it was originally done (see https://github.com/github/github-services/blob/master/.github/CONTRIBUTING.md for an example). I recall that being the case in the not too distant past.

I'm going off of https://help.github.com/articles/setting-guidelines-for-repository-contributors/, which states:

> To help your project contributors do good work, you can add a file with contribution guidelines to your project repository's root, docs, or .github folder.Then, whenever someone opens a pull request or creates an issue, they will see a link to that file.

**Pull request merged**